### PR TITLE
Force an encoding of utf-8 on liveinst installs (#1217504)

### PR DESCRIPTION
**Pull request by dashea**
Not saying we shouldn't fix the underlying problem, but this will make the ones we miss go away

### CODE REVIEW

**Comment by AdamWill** [comment]
This seems less hacky than the `reload(sys)` approach - and very similar to how we do it for non-lives, which is good. I'm very much +1 to this or *some* kind of 'force utf-8 for lives' approach - it's not strictly speaking the 'right thing to do' but in practice it's this or herd `UnicodeDecodeError`s until whenever we get anaconda to py3.

**Comment by AdamWill** [comment]
I checked this (by manually stuffing the changes into a live env before running liveinst) and it seems to do the trick - tested that a couple of the known bugs in Final TC1 - RHBZ#1217411 and RHBZ#1217610 - do not occur when the hack is in place.

**Comment by vpodzime** [comment]
I can hardly suprise anybody by saying that I like this. The only suggestion I have is that we should do the same on non-live installations (setting ``PYTHONPATH`` somewhere in the ``.service`` files we have) and drop the line from lorax that copies/moves the ``sitecustomize.py`` file to the right place in the ``install.img`` (``squashfs.img``).

**Comment by vpodzime** [comment]
Also note that this should be dropped with a switch to Python 3 (adding a label so that we don't forget to check it) or pushed only to the *f22-branch* with master being expected to do the switch right after F22 release.

File: anaconda.spec.in [hunk 1]
```diff
@@ -237,6 +237,10 @@ find %{buildroot} -type f -name "*.la" | xargs %{__rm}
 
 %ifarch %livearches
 desktop-file-install ---dir=%{buildroot}%{_datadir}/applications %{buildroot}%{_datadir}/applications/liveinst.desktop
+
+# Add a sitecustomize.py to be loaded by liveinst
+mkdir -p %{buildroot}%{_datadir}/anaconda/site-python
+install -m 0644 pyanaconda/sitecustomize.py %{buildroot}%{_datadir}/anaconda/site-python/
```
Comment (bcl) [inline anaconda.spec.in]: This should also be added to %files section.

Comment (vpodzime) [inline anaconda.spec.in]: > This should also be added to %files section.

AFAICT, we include everything from %{_datadir}/anaconda.

**Comment by dashea** [comment]
> Also note that this should be dropped with a switch to Python 3 (adding a label so that we don't forget to check it) or pushed only to the f22-branch with master being expected to do the switch right after F22 release.

I figured I'd only push it to f22-branch. We need to actually handle the string type disagreements since sitecustomize isn't going to handle cases like dirinstall or livemedia-creator, so I see this as just a temporary fix for F22 live.

File: anaconda.spec.in (hunk 1 above)
Comment (dashea) [inline anaconda.spec.in]: > > This should also be added to %files section.
> 
> AFAICT, we include everything from %{_datadir}/anaconda.

Yes, the directory ends up in anaconda-core without any extra help.

**Comment by dashea** [comment]
Pushed to f22-branch

**Comment by AdamWill** [comment]
:smile: :+1:

**Pull request closed without merging**